src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,147 @@
+            return true;
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The description of a todo cannot be empty.");
+            return false;
+        }
+    }
+
+    public static boolean addDeadline(String line, Task[] tasks, int taskNumber) {
+        String[] words = line.split(" ");
+        int index = 0;
+        String deadlineDescription = "";
+        String by = "";
+        for (int i = 0; i < words.length; i++) {
+            if (words[i].equals("/by")) {
+                index = i;
+                break;
+            }
+        }
+        if (index == 0) {
+            System.out.println("OOPS!!! The description of a deadline cannot be empty.");
+            return false;
+        }
+        for (int i = 1; i < index; i++) {
+            deadlineDescription = deadlineDescription + words[i] + " ";
+        }
+        for (int i = index + 1; i < words.length; i++) {
+            by = by + words[i] + " ";
+        }
+        tasks[taskNumber] = new Deadline(deadlineDescription, by);
+        taskNumber++;
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks[taskNumber - 1]);
+        System.out.println("Now you have " + taskNumber + " tasks in the list");
+        return true;
+    }
+
+    public static boolean addEvent(String line, Task[] tasks, int taskNumber) {
+        String[] words = line.split(" ");
+        int index = 0;
+        String eventDescription = "";
+        String at = "";
+        for (int i = 0; i < words.length; i++) {
+            if (words[i].equals("/at")) {
+                index = i;
+                break;
+            }
+        }
+        if (index == 0) {
+            System.out.println("OOPS!!! The description of a deadline cannot be empty.");
+            return false;
+        }
+        for (int i = 1; i < index; i++) {
+            eventDescription = eventDescription + words[i] + " ";
+        }
+        for (int i = index + 1; i < words.length; i++) {
+            at = at + words[i] + " ";
+        }
+        tasks[taskNumber] = new Event(eventDescription, at);
+        taskNumber++;
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks[taskNumber - 1]);
+        System.out.println("Now you have " + taskNumber + " tasks in the list");
+        return true;
+    }
+
+    public static void sayHi() {
+        System.out.println("Hello I'm duke.command.Duke");
+        System.out.println("What can I do for you?");
+    }
+
+    public static void sayBye() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    public static void running() {
+        Task[] tasks = new Task[100];
+        int taskNumber = 0;
+        String line;
+        Scanner in = new Scanner(System.in);
+        while (true) {
+            line = in.nextLine();
+            if (line.equals("bye")) {
+                return;
```

Comment:
In general, we do not use a return statement in a void function. If you want to break out of the current loop, use the "break" statement instead.